src/cachingreader.h

```diff
@@ -34,6 +35,8 @@ typedef struct Hint {
     int priority;
 } Hint;
 
+typedef QVarLengthArray<Hint, 512> HintVector;
```

Comment:
Ok, That sound reasonable. Can you drop some lines about it to the code. Thank you.